<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FillOptionsWidget</class>
 <widget class="QWidget" name="FillOptionsWidget">
  <property name="modal" stdset="0">
   <bool>false</bool>
  </property>
  <property name="windowModality">
   <enum>Qt::WindowModality::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>336</width>
    <height>304</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QGridLayout" name="gridLayout_2" columnstretch="6,10">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="2" column="0" colspan="2">
    <widget class="ColorPickerWidget" name="solid_color_picker" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>28</height>
      </size>
     </property>
     <property name="focusPolicy">
      <enum>Qt::FocusPolicy::NoFocus</enum>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The solid color for all glyphs.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="8" column="1">
    <widget class="Slider" name="sld_fill_radial_gradient_radius">
     <property name="minimum">
      <number>10</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="value">
      <number>55</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="RightAlignedLabel" name="lbl_fill_type">
     <property name="text">
      <string>Fill Type</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="Slider" name="sld_fill_radial_gradient_offset_x">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="focusPolicy">
      <enum>Qt::FocusPolicy::NoFocus</enum>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>9</number>
     </property>
     <item>
      <widget class="AngleDial" name="dial_fill_gradient_angle">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>35</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>35</width>
         <height>35</height>
        </size>
       </property>
       <property name="focusPolicy">
        <enum>Qt::FocusPolicy::NoFocus</enum>
       </property>
       <property name="styleSheet">
        <string notr="true">AngleDial
                                    {
                                    background-color: #27272B;
                                    color: #FFFFFF;
                                    }
                                </string>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="singleStep">
        <number>6</number>
       </property>
       <property name="wrapping">
        <bool>true</bool>
       </property>
       <property name="notchTarget">
        <double>45.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lbl_gradient_angle_value">
       <property name="text">
        <string>&lt;Angle&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="10" column="0">
    <widget class="RightAlignedLabel" name="lbl_fill_image">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Image</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="RightAlignedLabel" name="lbl_fill_radial_gradient_offset_x">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Offset X</string>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="RightAlignedLabel" name="lbl_fill_radial_gradient_radius">
     <property name="text">
      <string>Radius</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="RightAlignedLabel" name="lbl_fill_radial_gradient_offset_y">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Offset Y</string>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <widget class="ClickableLabel" name="lbl_fill_image_selection">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Click to load an image.</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Plain</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>:/logo.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="Slider" name="sld_fill_radial_gradient_offset_y">
     <property name="focusPolicy">
      <enum>Qt::FocusPolicy::NoFocus</enum>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="GradientEditorWidget" name="fill_gradient_picker" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>28</height>
      </size>
     </property>
     <property name="focusPolicy">
      <enum>Qt::FocusPolicy::NoFocus</enum>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The gradient for all glyphs. Primary click
                            to insert a new point. Secondary click to remove a point.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;
                        </string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="RightAlignedLabel" name="lbl_fill_gradient_angle">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Gradient Angle</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="ComboBox" name="cmb_fill_type">
     <item>
      <property name="text">
       <string>Solid Color</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Linear Gradient</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Radial Gradient</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Image</string>
      </property>
     </item>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RightAlignedLabel</class>
   <extends>QLabel</extends>
   <header>widgets/RightAlignedLabel.hpp</header>
  </customwidget>
  <customwidget>
   <class>ComboBox</class>
   <extends>QComboBox</extends>
   <header>widgets/ComboBox.hpp</header>
  </customwidget>
  <customwidget>
   <class>Slider</class>
   <extends>QSlider</extends>
   <header>widgets/Slider.hpp</header>
  </customwidget>
  <customwidget>
   <class>ColorPickerWidget</class>
   <extends>QWidget</extends>
   <header>widgets/ColorPickerWidget.hpp</header>
   <container>1</container>
   <slots>
    <signal>color_changed(QColor)</signal>
    <signal>on_color_changed(QColor)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>GradientEditorWidget</class>
   <extends>QWidget</extends>
   <header>widgets/GradientEditorWidget.hpp</header>
   <container>1</container>
   <slots>
    <signal>stops_changed(QGradientStops)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>ClickableLabel</class>
   <extends>QLabel</extends>
   <header>widgets/ClickableLabel.hpp</header>
   <slots>
    <signal>clicked()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>AngleDial</class>
   <extends>QDial</extends>
   <header>widgets/AngleDial.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/GeneralResources.qrc"/>
  <include location="../resources/UIResources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>solid_color_picker</sender>
   <signal>color_changed(QColor)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_solid_color_changed(QColor)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fill_gradient_picker</sender>
   <signal>stops_changed(QGradientStops)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_gradient_changed(QGradientStops)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dial_fill_gradient_angle</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_gradient_angle_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sld_fill_radial_gradient_offset_x</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_gradient_offset_x_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sld_fill_radial_gradient_offset_y</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_gradient_offset_y_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sld_fill_radial_gradient_radius</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_gradient_radius_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lbl_fill_image_selection</sender>
   <signal>clicked()</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_open_fill_image()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>229</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmb_fill_type</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>FillOptionsWidget</receiver>
   <slot>on_fill_type_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_open_fill_image()</slot>
  <slot>on_solid_color_changed(QColor)</slot>
  <slot>on_gradient_changed(QGradientStops)</slot>
  <slot>on_fill_type_changed()</slot>
  <slot>on_gradient_angle_changed(int)</slot>
  <slot>on_font_type_changed(int)</slot>
  <slot>on_font_size_changed(int)</slot>
  <slot>on_max_page_extent_changed(int)</slot>
  <slot>on_gradient_offset_x_changed(int)</slot>
  <slot>on_gradient_offset_y_changed(int)</slot>
  <slot>on_gradient_radius_changed(int)</slot>
 </slots>
</ui>
